octodns/provider/plan.py

```diff
@@ -67,19 +77,32 @@ def raise_if_unsafe(self):
            len(self.existing.records) >= self.MIN_EXISTING_RECORDS:
 
             existing_record_count = len(self.existing.records)
-            update_pcent = self.change_counts['Update'] / existing_record_count
-            delete_pcent = self.change_counts['Delete'] / existing_record_count
+            if existing_record_count > 0:
+                update_pcent = (self.change_counts['Update'] /
+                                existing_record_count)
+                delete_pcent = (self.change_counts['Delete'] /
+                                existing_record_count)
+            else:
+                update_pcent = 0
+                delete_pcent = 0
 
             if update_pcent > self.update_pcent_threshold:
-                raise UnsafePlan('Too many updates', update_pcent * 100,
-                                 self.update_pcent_threshold * 100,
-                                 self.change_counts['Update'],
-                                 existing_record_count)
+                raise TooMuchChange('Too many updates', update_pcent * 100,
+                                    self.update_pcent_threshold * 100,
+                                    self.change_counts['Update'],
+                                    existing_record_count)
             if delete_pcent > self.delete_pcent_threshold:
-                raise UnsafePlan('Too many deletes', delete_pcent * 100,
-                                 self.delete_pcent_threshold * 100,
-                                 self.change_counts['Delete'],
-                                 existing_record_count)
+                raise TooMuchChange('Too many deletes', delete_pcent * 100,
+                                    self.delete_pcent_threshold * 100,
+                                    self.change_counts['Delete'],
+                                    existing_record_count)
+
+        # If we have any changes of the root NS record for the zone it's a huge
+        # deal and force should always be required for extra care
+        if self.exists and [True for c in self.changes
+                            if c.record and c.record._type == 'NS' and
+                            c.record.name == '']:
```

Comment:
minor costmetic/readability: I like to use `any` in such cases:

```
change_ns = any(c.record and c.record._type == 'NS' and c.record.name == '' for c in self.changes)
```

Reply:
```suggestion
        if self.exists and any(c for c in self.changes
                               if c.record and c.record._type == 'NS' and
                               c.record.name == ''):
```